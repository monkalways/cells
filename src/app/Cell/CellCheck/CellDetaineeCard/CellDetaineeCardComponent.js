import React from 'react';
import PropTypes from 'prop-types';
import { Card, withStyles } from '@material-ui/core';

import {
  CellDetaineeCardHeader,
  CellDetaineeCardContent,
} from '../../../common/CellDetaineeCard';

import CardFooter from './CardFooter';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  cellName: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  cellCheck: PropTypes.shape({
    visual: PropTypes.bool.isRequired,
    verbal: PropTypes.bool.isRequired,
  }),
  onVisualClick: PropTypes.func,
  onVerbalClick: PropTypes.func,
};

const defaultProps = {
  cellCheck: null,
  onVisualClick: null,
  onVerbalClick: null,
};

export const CellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  cellName,
  isAuthenticated,
  cellCheck,
  onVisualClick,
  onVerbalClick,
}) => (
  <Card className={classes.card}>
    <CellDetaineeCardHeader cellDetainee={cellDetainee} />
    <CellDetaineeCardContent
      cellDetainee={cellDetainee}
      first="cells"
      second={cellName}
      isAuthenticated={isAuthenticated}
    />
    <CardFooter
      cellCheck={cellCheck}
      onVisualClick={onVisualClick}
      onVerbalClick={onVerbalClick}
    />
  </Card>
);

CellDetaineeCardComponent.propTypes = propTypes;
CellDetaineeCardComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
  },
}))(CellDetaineeCardComponent);
