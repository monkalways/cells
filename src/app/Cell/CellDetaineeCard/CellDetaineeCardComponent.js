import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardMedia, withStyles } from '@material-ui/core';

import CellDetaineeCardHeader from './CellDetaineeCardHeader';
import CellDetaineeCardFooter from './CellDetaineeCardFooter';
import detaineeImage from './detainee.PNG';
import detaineeAnonymousImage from './detainee-anonymous.PNG';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};

const CellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  isAuthenticated,
}) => {
  const getDetaineeImage = () => {
    if (isAuthenticated) {
      const { intakePhotoResourceUri } = cellDetainee;
      if (intakePhotoResourceUri) {
        return intakePhotoResourceUri;
      }
      return detaineeImage;
    }
    return detaineeAnonymousImage;
  };
  return (
    <Card className={classes.card}>
      <CellDetaineeCardHeader cellDetainee={cellDetainee} />
      <CardMedia
        className={classes.media}
        image={getDetaineeImage()}
        title={`${cellDetainee.firstName} ${cellDetainee.lastName}`}
      />
      <CellDetaineeCardFooter
        cellDetainee={cellDetainee}
        isAuthenticated={isAuthenticated}
      />
    </Card>
  );
};

CellDetaineeCardComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: '#EBEBEB',
  },
  media: {
    height: 0,
    paddingTop: '66.25%', // 16:9
  },
}))(CellDetaineeCardComponent);
