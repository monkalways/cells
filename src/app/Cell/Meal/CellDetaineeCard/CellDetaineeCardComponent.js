import React from 'react';
import PropTypes from 'prop-types';
import { Card, withStyles } from '@material-ui/core';

import {
  CellDetaineeCardHeader,
  CellDetaineeCardContent,
} from '../../../common/CellDetaineeCard';

import CardFooter from './CardFooter';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  cellName: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  meal: PropTypes.shape({
    accept: PropTypes.bool.isRequired,
    reject: PropTypes.bool.isRequired,
    notApplicable: PropTypes.bool.isRequired,
  }),
  onAcceptClick: PropTypes.func.isRequired,
  onRejectClick: PropTypes.func.isRequired,
  onNotApplicableClick: PropTypes.func.isRequired,
};

const defaultProps = {
  meal: null,
};

export const CellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  cellName,
  isAuthenticated,
  meal,
  onAcceptClick,
  onRejectClick,
  onNotApplicableClick,
}) => (
  <Card className={classes.card}>
    <CellDetaineeCardHeader cellDetainee={cellDetainee} />
    <CellDetaineeCardContent
      cellDetainee={cellDetainee}
      first="cells"
      second={cellName}
      isAuthenticated={isAuthenticated}
    />
    <CardFooter
      meal={meal}
      onAcceptClick={onAcceptClick}
      onRejectClick={onRejectClick}
      onNotApplicableClick={onNotApplicableClick}
    />
  </Card>
);

CellDetaineeCardComponent.propTypes = propTypes;
CellDetaineeCardComponent.defaultProps = defaultProps;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
  },
}))(CellDetaineeCardComponent);
