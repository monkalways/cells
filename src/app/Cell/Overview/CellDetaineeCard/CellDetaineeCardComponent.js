import React from 'react';
import PropTypes from 'prop-types';
import { Card, withStyles } from '@material-ui/core';

import {
  CellDetaineeCardHeader,
  CellDetaineeCardContent,
} from '../../../common/CellDetaineeCard';

import CardFooter from './CardFooter';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetainee: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
  }).isRequired,
  cellName: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
};

const OverviewCellDetaineeCardComponent = ({
  classes,
  cellDetainee,
  cellName,
  isAuthenticated,
}) => (
  <Card className={classes.card}>
    <CellDetaineeCardHeader cellDetainee={cellDetainee} />
    <CellDetaineeCardContent
      cellDetainee={cellDetainee}
      first="cells"
      second={cellName}
      isAuthenticated={isAuthenticated}
    />
    <CardFooter cellDetainee={cellDetainee} isAuthenticated={isAuthenticated} />
  </Card>
);

OverviewCellDetaineeCardComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  card: {
    maxWidth: theme.spacing.unit * 50,
    backgroundColor: theme.palette.background.paper,
  },
}))(OverviewCellDetaineeCardComponent);
