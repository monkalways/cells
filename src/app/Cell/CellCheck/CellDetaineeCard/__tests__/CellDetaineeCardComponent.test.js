import React from 'react';
import { shallow } from 'enzyme';

import { CellDetaineeCardComponent } from '../CellDetaineeCardComponent';

describe('CellDetaineeCardComponent', () => {
  const setup = () => {
    const classes = {
      card: 'card',
    };
    return shallow(<CellDetaineeCardComponent
      classes={classes}
      cellDetainee={{}}
      cellName="c1"
      isAuthenticated={false}
      cellCheck={{}}
      onVisualClick={jest.fn()}
      onVerbalClick={jest.fn()}
    />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
