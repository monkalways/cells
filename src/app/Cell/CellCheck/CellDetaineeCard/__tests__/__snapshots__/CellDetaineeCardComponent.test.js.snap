// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellDetaineeCardComponent should render 1`] = `
<WithStyles(Card)
  className="card"
>
  <WithStyles(CellDetaineeCardHeaderComponent)
    cellDetainee={
      Object {
        "firstName": "John",
        "lastName": "Smith",
      }
    }
  />
  <WithStyles(withRouter(CellDetaineeCardContentComponent))
    cellDetainee={
      Object {
        "firstName": "John",
        "lastName": "Smith",
      }
    }
    first="cells"
    isAuthenticated={false}
    second="c1"
  />
  <WithStyles(CardFooterComponent)
    cellCheck={
      Object {
        "verbal": false,
        "visual": true,
      }
    }
    onVerbalClick={[MockFunction]}
    onVisualClick={[MockFunction]}
  />
</WithStyles(Card)>
`;
